src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -150,30 +150,28 @@ def get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood=None, release
     except Exception as e:
         telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
                                 summary='Error while fetching helm chart registry path')
-        raise CLIError("Error while fetching helm chart registry path: " + str(e))
+        raise CLIInternalError("Error while fetching helm chart registry path: " + str(e))
     if r.content:
         try:
             return r.json().get('repositoryPath')
         except Exception as e:
             telemetry.set_exception(exception=e, fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
                                     summary='Error while fetching helm chart registry path')
-            raise CLIError("Error while fetching helm chart registry path from JSON response: " + str(e))
+            raise CLIInternalError("Error while fetching helm chart registry path from JSON response: " + str(e))
     else:
         telemetry.set_exception(exception='No content in response', fault_type=consts.Get_HelmRegistery_Path_Fault_Type,
                                 summary='No content in acr path response')
-        raise CLIError("No content was found in helm registry path response.")
+        raise CLIInternalError("No content was found in helm registry path response.")
 
 
 def arm_exception_handler(ex, fault_type, summary, return_if_not_found=False):
     if isinstance(ex, AuthenticationError):
-        telemetry.set_user_fault()
         telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
-        raise CLIError("Authentication error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
+        raise ClientRequestError("Authentication error occured while making ARM request: " + str(ex) + "\nSummary: {}".format(summary))
```

Comment:
ClientRequestError is incorrect. It means client was not able to send the request to ARM. https://github.com/Azure/azure-cli/blob/119856d4f5984e5a4ef0cdb5bc233c6f07da2720/src/azure-cli-core/azure/cli/core/azclierror.py#L205

Maybe use 'UnauthorizedError' here